Index: git/arch/nios2/include/asm/io.h
===================================================================
--- git.orig/arch/nios2/include/asm/io.h
+++ git/arch/nios2/include/asm/io.h
@@ -208,6 +208,11 @@ static inline void io_insl(unsigned int
 #define iowrite16(val, X)	writew(val, X)
 #define iowrite32(val, X)	writel(val, X)
 
+#define ioread16be(X)		be16_to_cpu(ioread16(X))
+#define ioread32be(X)		be32_to_cpu(ioread32(X))
+#define iowrite16be(val, X)	iowrite16(be16_to_cpu(val), (X))
+#define iowrite32be(val, X)	iowrite32(be32_to_cpu(val), (X))
+
 #ifdef CONFIG_MMU
 
 extern void __iomem *__ioremap(unsigned long physaddr, unsigned long size,
